capa/features/extractors/viv/global_.py

```diff
@@ -1,20 +1,18 @@
 import logging
 from typing import Tuple, Iterator
 
-import envi.archs.i386
-import envi.archs.amd64
-
 from capa.features.common import ARCH_I386, ARCH_AMD64, Arch, Feature
 from capa.features.address import NO_ADDRESS, Address
 
 logger = logging.getLogger(__name__)
 
 
 def extract_arch(vw) -> Iterator[Tuple[Feature, Address]]:
-    if isinstance(vw.arch, envi.archs.amd64.Amd64Module):
+    arch = vw.getMeta("Architecture")
+    if arch == "amd64":
         yield Arch(ARCH_AMD64), NO_ADDRESS
 
-    elif isinstance(vw.arch, envi.archs.i386.i386Module):
+    elif arch == "i386":
```

Comment:
this is much nicer!

Reply:
^^ seems to be what Willi describes below, need to discuss if we can update or if there's still something off in vivisect

Reply:
if you're seeing `loadModuleFromPathTup` then i think you're still using an old viv release, since that routine is no longer present in viv master